workspaces/core/src/select/Select.tsx

```diff
@@ -0,0 +1,150 @@
+import React, {forwardRef, useState} from "react";
+import ChevronDown from "../private/icons/ChevronDown";
+import ChevronUp from "../private/icons/ChevronUp";
+import Clear from "../private/icons/Clear";
+import {determineMenuBackgroundColor} from "./private/getBackgroundColor";
+import {getWrapperClasses} from "./private/getWrapperClasses";
+import Option from "./private/types/OptionProps";
+import SelectProps from "./private/types/SelectProps";
+import useEventListeners from "./private/useEventListeners";
+
+const Select = forwardRef<HTMLDivElement, SelectProps>(({
+  options,
+  label,
+  value,
+  disabled,
+  isError,
+  placeholderSlot,
+  placeholderValue,
+  hintSlot,
+  onChange,
+  size = 'md',
+  amountOfVisibleItems = 9999,
+  menuWidth,
+  headerSlot,
+  footerSlot,
+  leftSlot,
+  isSharpLeftSide,
+  isSharpRightSide,
+  isSharpTopSide,
+  isSharpBottomSide,
+  isTopBottomBorderHidden,
+  isSideBorderHidden,
+  isRtl,
+  isSearchable,
+  noResultsMessage,
+  formatOptionLabel
+}) => {
+  const [hoveredIndex, setHoveredIndex] = useState(-1);
+  const [selectedIndex, setSelectedIndex] = useState(-1);
+  const placeholder = value ? options?.filter((option) => option.value === value)[0]?.label
+    : placeholderSlot ?? placeholderValue ?? 'Choose an option';
+  const {
+    menuOpen,
+    search,
+    menuOptions,
+    inputFocused,
+    inputRef,
+    selectRef,
+    menuRef,
+    onSelectClick,
+    onInputFocus,
+    selectMenuItem,
+    handleInputKeydown
+  } = useEventListeners({
```

Comment:
could you please give the hook a more specific name, like useSelect(), (as this hook contain the logic of this particular component),  because  useEventListeners is the too common name